### DISCUSSION TITLE
Related: ListsEnable

### DISCUSSION THREAD

**Discussion by sez11a** [lervag/wiki.vim#353] [Q&A] [answered]
I know this is a different repo, but it's related (I wouldn't be using lists if not for wiki.vim). It also feels like a "stupid" question. I'm trying to enable lists for all Markdown files, using `lazy.nvim`, but it's not working. Is there something wrong with my syntax? 

```lua
  {
  'lervag/lists.vim',
    init = function()
      vim.g.lists_filetypes = {
          'markdown',
          'mkd'
      }
    end
  },
```

**Comment by lervag** [comment]
I think the problem may be that I've used a stupid name for the option. The list is a list of file extensions, not a list of filetypes. Thus, if your markdown files are named `foo.md`, then you should add `'md'` to that list.

**Reply by sez11a** [reply]
Hm; even if I make it a single entry of just `'md'`, it's still not working. Full steps: 

1. `<leader>w<leader>w` to open today's journal. 
2. Scroll down to item in list. 
3. `<c-s>` to complete the TODO; nothing happens. 
4. Type :ListsEnable
5. `<c-s>` on same item; it's marked complete.

**Reply by lervag** [reply, marked as answer]
Are you using this config now?

```vim
{
  'lervag/lists.vim',
  init = function()
    vim.g.lists_filetypes = { 'md', 'mkd' }
  end
},
```

After doing `<leader>w<leader>w` - what's the output of `:autocmd lists`?

**Reply by sez11a** [reply]
Yes; that's the config. When I try `:autocmd lists`, I get 

`E216: No such group or event: lists`

Just for completeness, here's the code (only difference from yours above is in formatting): 

```lua
  {
  'lervag/lists.vim',
    init = function()
      vim.g.lists_filetypes = {
          'md',
          'mkd'
      }
    end
  },
```

EDIT: Interestingly, no, there was a difference. I had only one item in the array: `md`. That didn't work. It seems to be working now that I have two items. Weird.

**Reply by lervag** [reply]
Yes, that is very strange - I'll look into it. Notice, though: it _may_ be important to use `init` instead of `config` here. I noticed in the first version of this reply, you used `config = ...`.

**Reply by lervag** [reply]
I've opened a discussion on this topic (of `init` vs `config`): https://github.com/folke/lazy.nvim/discussions/1427

**Reply by lervag** [reply]
Notice, I can't reproduce the problem with only having a single element in the list. I believe your problem may have been related to the `config` vs `init` thing..?

**Reply by sez11a** [reply]
I had changed it from `init` to `config` during my testing, and then changed it back to `init` to follow your code. But every other `lazy.nvim` example I can see uses `config`. It definitely wasn't working initially using `init` either.

**Reply by lervag** [reply]
> I had changed it from `init` to `config` during my testing, and then changed it back to `init` to follow your code. But every other `lazy.nvim` example I can see uses `config`.

Yes, which is why I've opened the discussion at lazy.nvim. I think the load order when you use the `config` option is unexpected here. Regardless, it is perfectly safe to use the `init` key, so just do that.

> It definitely wasn't working initially using `init` either.

No, but that was what I addressed in my very first reply, right?
